### ISSUE TITLE
Rotation 90 kobo glo hd

### ISSUE THREAD

**Issue by takov751** [NiLuJe/FBInk#66]
Display rotation error. Fbink just ignores any rotatation I set.failed miserably.
```
[root@(none) gotest]# echo 1 >>/sys/class/graphics/fb0/rotate
[root@(none) gotest]# fbink -x 1 -y 10 "Hello World!"
[FBInk] Detected a Kobo Glo HD (371 => Alyssum @ Mark 6)
[FBInk] Clock tick frequency appears to be 100 Hz
[FBInk] Screen density set to 300 dpi
[FBInk] Variable fb info: 1072x1448, 16bpp @ rotation: 1 (Clockwise, 90°)
[FBInk] Fixed fb info: ID is "mxc_epdc_fb", length of fb mem: 6782976 bytes & line length: 2176 bytes
[FBInk] Canonical rotation: 2 (Upside Down, 180°)

```
I am writing a small go application which would communicate with an REST-API. Like a streamdeck, just with kobo. might have to render text as image on api end

**Comment by NiLuJe** [comment]
You're stuck in the crappy pickel and/or boot fb layout, don't do that, it's awful.

Use `fbdepth -d 8 -R UR` to switch to a sane layout.

**Comment by takov751** [comment]
I am not running nickel and I am on the new 4+ firmware. I am running KSM 09, however I will look into it

**Comment by NiLuJe** [comment]
Yep, that would be KSM. In which case, witching the layout *will* break KSM.

**Comment by takov751** [comment]
Is there any other menu you would recommend or should I just kill ksm when app starts and then when app exits reboot device ?

**Comment by NiLuJe** [comment]
KSM is an unsupported configuration in and of itself, and is pretty much incompatible with with basically everything else, because it does things the slightly wrong way, given its history of following old nickel versions doing broken things.

TL;DR: Yeah, get rid of it. Whatever else you might want to use instead is your own decision, though ;).
